cxx/emissions/sometimes.hh

```diff
@@ -1,50 +1,69 @@
 #pragma once
 
+#include <cmath>
 #include <unordered_map>
 
 #include "distributions/beta_bernoulli.hh"
 #include "emissions/base.hh"
 
-// An Emission class that sometimes applies BaseEmissor and sometimes doesn't.
-// BaseEmissor must assign zero probability to <clean, dirty> pairs with
-// clean == dirty.  [For example, BitFlip and Gaussian both satisfy this].
-template <typename BaseEmissor>
-class Sometimes : public Emission<typename std::tuple_element<
-                      0, typename BaseEmissor::SampleType>::type> {
+// An Emission class that sometimes applies another Emission and sometimes
+// doesn't.
+template <typename SampleType = double>
+class Sometimes : public Emission<SampleType> {
  public:
-  using SampleType =
-      typename std::tuple_element<0, typename BaseEmissor::SampleType>::type;
   BetaBernoulli bb;
-  BaseEmissor be;
+  Emission<SampleType>* be;  // We own be.
+  bool can_dirty_equal_clean;
```

Comment:
optional: this is more readable to me when phrased as a statement (e.g. `dirty_can_equal_clean`) rather than a question

Reply:
Done.